mkosi/distributions/gentoo.py

```diff
@@ -563,20 +441,4 @@ def kernel_image(name: str, architecture: str) -> Path:
     @classmethod
     @complete_step("Installing Gentoo…")
     def install(cls, state: "MkosiState") -> None:
-        # this will fetch/fix stage3 tree and portage confgired for mkosi
-        gentoo = Gentoo(state)
-
-        if gentoo.pkgs_fs:
-            gentoo.invoke_emerge(state, pkgs=gentoo.pkgs_fs)
-
-        if not state.do_run_build_script and state.config.bootable:
-            # The gentoo stage3 tarball includes packages that may block chosen
-            # pkgs_boot. Using Gentoo.EMERGE_UPDATE_OPTS for opts allows the
-            # package manager to uninstall blockers.
-            gentoo.invoke_emerge(state, pkgs=gentoo.pkgs_boot, opts=Gentoo.EMERGE_UPDATE_OPTS)
-
-        if state.config.packages:
-            gentoo.invoke_emerge(state, pkgs=state.config.packages)
-
-        if state.do_run_build_script:
-            gentoo.invoke_emerge(state, pkgs=state.config.build_packages)
+        Gentoo(state)
```

Comment:
It looks like you're doing all the work in the `__init__` method of this class. At this point it doesn't need to be a class anymore. You could just move all of it into this method, no?

Reply:
one at the time :) moving things around like that is yet another big change.

Reply:
is this necessary, `fedora` seems to be doing the same thing, albeit not inside a class.